### PR TITLE
Siva | BAH-950 | filter column in event_log table is not updated to correct value in Bahmni Connect

### PR DESCRIPTION
**Pull request by sivareddyp**
Open MRS Talk Thread - https://talk.openmrs.org/t/filter-column-in-event-log-table-is-not-updated-to-correct-value/25657

### CODE REVIEW

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java [hunk 1]
```diff
@@ -184,4 +204,36 @@ private boolean isOfflineConceptEvent(String eventUuid) {
         final Concept offlineConcept = conceptService.getConceptByName("Offline Concepts");
         return offlineConcept != null && offlineConcept.getSetMembers().contains(concept);
     }
+
+    protected String getFilterForLocalIdentifierSource(String identifier) {
+        Session session = null;
+        try {
+            session = sessionFactory.getCurrentSession();
+            session.beginTransaction();
```
Comment (sumanmaity112) [inline omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java:212]: Do we really need a transaction as we are reading only?

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java [hunk 2]
```diff
@@ -184,4 +204,36 @@ private boolean isOfflineConceptEvent(String eventUuid) {
...
+            String prefix = session.doReturningWork(new ReturningWork<String>() {
+                @Override
+                public String execute(Connection conn) throws SQLException {
+                    PreparedStatement pStmt = null;
+                    try {
+                        pStmt = conn.prepareStatement(identifierPrefixQuery);
+                        pStmt.setString(1, identifier);
+                        ResultSet resultSet = pStmt.executeQuery();
+                        if(resultSet.next())
```
Comment (sumanmaity112) [inline omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java]: I think here we can use ternary condition

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java [hunk 3]
```diff
@@ -184,4 +204,36 @@ private boolean isOfflineConceptEvent(String eventUuid) {
...
+                    finally {
+                        pStmt.close();
+                    }
+                }
+            });
+            session.getTransaction().commit();
+            return prefix;
+        }
+        catch(HibernateException e) {
+            RuntimeException exception = new RuntimeException("Error while executing query for getting local identifier [" + identifier + "]");
```
Comment (sumanmaity112) [inline omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java:232]: @sivareddytw It'll be better if we log the actual error as well

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java [hunk 4]
```diff
@@ -50,6 +53,13 @@
     private PlatformTransactionManager platformTransactionManager;
     @Mock
     private LocationService locationService;
+    @Mock
+    private SessionFactory sessionFactory;
+    @Mock
+    Session session;
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java]: Please define the scope of this veriable

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java [hunk 5]
```diff
@@ -50,6 +53,13 @@
...
+    @Mock
+    Transaction transaction;
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java]: same as before

**Review by sumanmaity112 — CHANGES_REQUESTED** [review]

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java (hunk 5 above)
Comment (sivareddyp) [reply omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java]: Added

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java (hunk 2 above)
Comment (sivareddyp) [reply omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java]: Updated...

File: omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java [hunk 6]
```diff
@@ -69,6 +79,11 @@ public void setUp() throws Exception {
         List registeredComponents = new ArrayList();
         registeredComponents.add(platformTransactionManager);
         Mockito.when(Context.getRegisteredComponents(PlatformTransactionManager.class)).thenReturn(registeredComponents);
+        List sessionFactoryComponents = new ArrayList();
```
Comment (sumanmaity112) [inline omod/src/test/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategyTest.java]: Can we add type definition. Example: `new ArrayList<String>() `

File: omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java (hunk 1 above)
Comment (sivareddyp) [reply omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java:212]: Creating transaction boundaries are always necessary even for read-only operations. No communication with the database can occur outside of a database transaction. If we do not end (i.e commit or rollback) a transaction , the database connection used by this transaction will not be released and your database will run out of the available connections eventually.

**Review by sumanmaity112 — APPROVED** [review]

**Review by binduak — APPROVED** [review]

**Pull request merged**